client/my-sites/site-settings/index.js

```diff
@@ -111,5 +112,13 @@ export default function () {
 	page( '/settings/analytics/:site_id?', redirectToTraffic );
 	page( '/settings/seo/:site_id?', redirectToTraffic );
 
-	page( '/settings/:section', legacyRedirects, siteSelection, sites, makeLayout, clientRender );
+	page(
+		'/settings/:section',
+		legacyRedirects,
+		siteSelection,
+		redirectToGeneral,
```

Comment:
I'd recommend combining the logic for redirecting to general settings with and without site in the same function. Meaning, combining `redirectToGeneral` with https://github.com/Automattic/wp-calypso/blob/079c299b82810f05859268367a198dac20141c46/client/my-sites/site-settings/index.js#L21 if that makes sense. In other words, redirecting to `/general` would ideally happen in one place, not in several places.